### DISCUSSION TITLE
LSD2: Signal Segmentation Fault while running time tree search

### DISCUSSION THREAD

**Discussion by GiorgioMuneretto** [iqtree/iqtree2#194] [Q&A] [answered]
I am running the following command:

```
iqtree -s concatenated.out -nt AUTO --date DATE_FILE -te FigTree_old_tree_clustered_fixed2.nwk --date-tip 0 --date-root -520 -p partitions.txt

```
and this is the error I get:

```
Building time tree by least-square dating (LSD) with command:
lsd -i partitions.txt.timetree.subst -s 3511 -c -o partitions.txt.timetree -r a -d partitions.txt.timetree.date 

TREE 1
*PROCESSING:
Reading the tree ... 
Collapse 0 (over 153) internal branches having branch length <= 0.000 (settable via option -l)
ERROR: STACK TRACE FOR DEBUGGING:
ERROR: 1   funcAbort()
ERROR: 2   ()
ERROR: 3   isAncestor(Node**, int, int)
ERROR: 4   mrca(Node**, std::vector<int, std::allocator<int> >)
ERROR: 5   initConstraint(Pr*, Node**)
ERROR: 6   lsd::buildTimeTree(int, char**, lsd::InputOutputStream*)
ERROR: 7   runLSD2(PhyloTree*)
ERROR: 8   doTimeTree(PhyloTree*)
ERROR: 9   runPhyloAnalysis(Params&, Checkpoint*)
ERROR: 10   main()
ERROR: 11   __libc_start_main()
ERROR: 12   ()
ERROR: 
ERROR: *** IQ-TREE CRASHES WITH SIGNAL SEGMENTATION FAULT
ERROR: *** For bug report please send to developers:
ERROR: ***    Log file: partitions.txt.log
ERROR: ***    Alignment files (if possible)
```
What is the problem?

Thank you,
Giorgio

[partitions.txt](https://github.com/iqtree/iqtree2/files/15284384/partitions.txt)
[concatenated.out](https://github.com/GiorgioMuneretto/Files/blob/228b020c5ec6618393d1a452235a9d9297b3d230/project_subunits/concatenated.out)

**Comment by tothuhien** [comment]
Hi Giorgio,
Thanks for reporting the issue. I was trying to detect the error but could not download the concatenated.out file. Maybe something with permission restriction? Could you upload the file on this discussion channel?  It's also useful to have other files (DATE_FILE, FigTree_old_tree_clustered_fixed2.nwk) that you used as input.
Cheers,
Hien

**Comment by GiorgioMuneretto** [comment]
Dear Hien,

Thank you very much for paying attention to the issue! Below you can find the other three files. Note that concatenated.out and FigTree_old_tree_clustered_fixed2.nwk were converted into txt files (in order to attach them here).

[DATE_FILE.txt](https://github.com/iqtree/iqtree2/files/15347478/DATE_FILE.txt)
[concatenated.out.txt](https://github.com/iqtree/iqtree2/files/15347516/concatenated.out.txt)
[FigTree_old_tree_clustered_fixed2.nwk.txt](https://github.com/iqtree/iqtree2/files/15347569/FigTree_old_tree_clustered_fixed2.nwk.txt)

Let me know!

Giorgio

**Comment by tothuhien** [comment, marked as answer]
Hi Giorgio, 
Sorry for the late response, I'm just back from vacation. Thanks for providing the data. I've tried to rerun your analysis with the exact same command, but I didn't have any error. I got the iqtree as well as the final dated tree as in the attachment. The log file is also in attachment.
[partitions.txt.treefile.txt](https://github.com/iqtree/iqtree2/files/15385819/partitions.txt.treefile.txt)
[partitions.txt.timetree.nex.txt](https://github.com/iqtree/iqtree2/files/15385821/partitions.txt.timetree.nex.txt)
[partitions.txt.log](https://github.com/iqtree/iqtree2/files/15385842/partitions.txt.log)

I used iqtree version 2.3.4. What version did you use?
Best, 
Hien

**Reply by GiorgioMuneretto** [reply]
Thank you very much Hien, that solved the problem!
I was using version 2.0 and upon upgrade to 2.3.3 it worked!
Best regards,
Giorgio
